internal/controllers/device_data_controller_v2.go

```diff
@@ -0,0 +1,153 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/DIMO-Network/devices-api/pkg/grpc"
+	"github.com/DIMO-Network/shared"
+	"github.com/gofiber/fiber/v2"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/DIMO-Network/device-data-api/internal/config"
+	"github.com/DIMO-Network/device-data-api/internal/services"
+)
+
+type DeviceDataControllerV2 struct {
```

Comment:
Other `v2` endpoints are not using this controller, such as `GetVehicleStatusV2` and `GetHistoricalPermissionedV2`. Those should be migrated, or this should be renamed.

Reply:
We tend to make very congested controllers, if we are making v2 changes, this should be in a new controller rather than batching it together with the v1 and ending up congesting that file. So `GetVehicleStatusV2` should have been in its own controller, rather than moving it into the existing controller for v1. 

Note: this was done with Tesla Fleet API which is a v2 version as well, for thesame reason to avoid congestion.

Reply:
I will move the mentioned controller functions as suggested